src/ResourceManager/Common/Commands.Common.Authentication.ResourceManager/AzureRmProfile.cs

```diff
@@ -577,7 +578,7 @@ public bool TryCopyProfile(AzureRmProfile other)
                 TrySetContext(context.Key, context.Value);
             }
 
-            if (other.DefaultContext != null)
+            if (other.DefaultContext != null && this.DefaultContext != other.DefaultContext)
```

Comment:
Will have to ensure that the default equality comparer does the right thing in this case (i.e. that it can handle differences in context names).  I wonder, if the DefaultContextKey has been set above, and the value is not null, whether this will evaluate to true.